extensions/amp-bind/0.1/test/test-bind-expression.js

```diff
@@ -30,534 +30,599 @@ describe('BindExpression', () => {
+      expect(evaluate('{a: "b", c: "d",}')).to.deep.equal({a: 'b', c: 'd'});
+    });
+
+    it('should evaluate undefined vars and properties to null', () => {
+      expect(evaluate('foo')).to.be.null;
+      expect(evaluate('foo.bar')).to.be.null;
+      expect(evaluate('foo["bar"]')).to.be.null;
+      expect(evaluate('foo[bar]')).to.be.null;
+      expect(evaluate('foo[0]')).to.be.null;
+    });
+  });
+
+  describe('function invocation', () => {
+    it('Math functions', () => {
+      expect(evaluate('abs(-1)')).to.equal(1);
+      expect(evaluate('ceil(0.1)')).to.equal(1);
+      expect(evaluate('floor(1.9)')).to.equal(1);
+      expect(evaluate('max(0, 1)')).to.equal(1);
+      expect(evaluate('min(0, 1)')).to.equal(0);
+      expect(evaluate('round(0.6)')).to.equal(1);
+      const r = evaluate('random()');
+      expect(r).to.be.at.least(0);
+      expect(r).to.be.at.below(1);
+      expect(evaluate('sign(-1)')).to.equal(-1);
+
+      // Functions should not conflict with scope variables.
+      expect(evaluate('abs(-2) + abs', {abs: 2})).to.equal(4);
+
+      // Don't support non-whitelisted functions.
+      expect(() => {
+        evaluate('sin(0.5)');
+      }).to.throw(unsupportedFunctionError);
+      expect(() => {
+        evaluate('pow(3, 2)');
+      }).to.throw(unsupportedFunctionError);
+
+      // Don't support calling functions with `Math.` prefix.
+      expect(() => {
+        evaluate('Math.abs(-1)', {Math});
+      }).to.throw(unsupportedFunctionError);
+    });
+
+    it('encodeURI and encodeURIComponent', () => {
+      expect(evaluate('encodeURI("http://www.google.com/s p a c e.html")'))
+          .to.equal('http://www.google.com/s%20p%20a%20c%20e.html');
+      expect(evaluate('encodeURIComponent("http://www.google.com/foo?foo=bar")'))
+          .to.equal('http%3A%2F%2Fwww.google.com%2Ffoo%3Ffoo%3Dbar');
+    });
+
+    it('copyAndSplice()', () => {
+      const a = [1, 2, 3];
+      expect(() => evaluate('copyAndSplice()')).to.throw(/not an array/);
+      expect(() => evaluate('copyAndSplice(x)', {x: 8472}))
+          .to.throw(/not an array/);
+      expect(evaluate('copyAndSplice(a)', {a})).to.not.equal(a);
+      expect(evaluate('copyAndSplice(a)', {a})).to.deep.equal(a);
+      expect(evaluate('copyAndSplice(a, 1)', {a})).to.deep.equal([1]);
+      expect(evaluate('copyAndSplice(a, 1, 1)', {a})).to.deep.equal([1, 3]);
+      expect(evaluate('copyAndSplice(a, 1, 1, 47)', {a}))
+          .to.deep.equal([1, 47, 3]);
+    });
+
+    it('return null when caller is null', () => {
+      expect(evaluate('null.charAt(0)')).to.be.null;
+      expect(evaluate('null.includes(3)')).to.be.null;
+      expect(evaluate('a.indexOf("baz")', {a: null})).to.be.null;
+      expect(evaluate('(false || null).replace("foo", "bar")')).to.be.null;
+    });
+
+    it('disallow: function declarations', () => {
+      expect(() => { evaluate('function() {}'); }).to.throw();
```

Comment:
This is illegal in JS syntax, anyways. You'd need to make it an expression: `'(function() {})'`.